docs: Federated Auth bundle Uber JAR (#853)
awslabs/aws-advanced-jdbc-wrapper: docs/using-the-jdbc-driver/using-plugins/UsingTheFederatedAuthPlugin.md

> [!WARNING]
> To preserve compatibility with customers using the community driver, this plugin requires the [AWS Java SDK RDS v2.7.x](https://central.sonatype.com/artifact/software.amazon.awssdk/rds) and the [AWS Java SDK STS v2.7.x](https://central.sonatype.com/artifact/software.amazon.awssdk/sts) to be included separately in the classpath. The AWS Java SDK RDS and AWS Java SDK STS are runtime dependencies and must be resolved.

### Bundled Uber JAR
Included in AWS JDBC Driver release, is an Uber JAR that bundles the AWS JDBC Driver and all the package dependencies needed to use the Federated Authentication Plugin. 
It is suffixed with `-bundle-federated-auth`.

> This JAR is a drop-in ready solution intended for users who do not have an automated package manager like Maven or Gradle. 

Reviewer: ```suggestion
This JAR is a drop-in ready solution and is **recommended for customers who do not have an automated package manager like Maven or Gradle**. 
```

> As this plugin has a number of transitive dependencies, the goal of this JAR is to eliminate the need for such users to manually source all the dependencies and avoid the headaches of managing them. 

Reviewer: ```suggestion
As this plugin has a number of transitive dependencies, the goal of this JAR is to eliminate the need to manually source all the dependencies and avoid potential issues with managing them. 
```

In that spirit, the dependencies in this JAR are shaded with the prefix `shaded` to avoid potential package conflicts with pre-existing packages in your environment.

> It is important to note that the bundled Uber JAR quite large and exceeds 15 MB. So please take that into account when deciding if this solution is for you.

Reviewer: ```suggestion
It is important to note that the bundled Uber JAR is larger (exceeds 15MB) than our `aws-advanced-jdbc-wrapper-X.X.X.jar`. So please take that into account when deciding if this solution is for you.
```
